sdk/spring-sdk/src/main/java/kalix/springsdk/annotations/GenerateEntityKey.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kalix.springsdk.annotations;
+
+import java.lang.annotation.*;
+
+/**
+ * Indicates that Kalix should generate an entity key when this method is invoked.
+
+ * The generated entity key will be accessible via the entity context supplied in the SDK.
+ * A method annotated with this annotations should not be annotated with {@link EntityKey},
+ * if it does, an error will be raised.
+ *
+ * The generated key will be a Version 4 (random) UUID. The UUID will be generated using a cryptographically secure
+ * random number generator.
+ */
+@Target(ElementType.METHOD)
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+public @interface GenerateEntityKey {
```

Comment:
The documentation for this quite hidden in the gRPC SDK, I think we should add it on the entities pages. 

I'm planning to do it on follow up PR and then also cover Value Entities.

Reply:
would you like to add the reference to an issue before closing this PR?